pkg/daemon/bootc.go

```diff
@@ -0,0 +1,245 @@
+func (s *Status) GetBootedImage() *BootEntry {
+	return s.Booted
+}
+
+// GetStagedImage finds the staged image for the next boot, or returns nil if none is found.
+func (s *Status) GetStagedImage() *BootEntry {
+	return s.Staged
+}
+
+// GetRollbackImage finds the rollback deployment, or returns nil if none is found.
+func (s *Status) GetRollbackImage() *BootEntry {
+	return s.Rollback
+}
+
+// GetBaseChecksum returns the ostree commit used as a base.
+func (b *BootEntry) GetBaseChecksum() string {
+	return b.Ostree.Checksum
+}
+
+// GetContainerImageReference returns the the container image reference.
+func (b *BootEntry) GetContainerImageReference() string {
+	return b.Image.Image.Image
+}
+
+// imgref.go
+// ImageReference captures an image signature verification policy alongside an image reference.
+// This mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L69
+type ImageReference struct {
+	Image     string          `json:"image"`
+	Transport string          `json:"transport"`
+	Signature *ImageSignature `json:"signature,omitempty"`
+}
+
+// ImageSignature mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L57
+// Same as https://docs.rs/ostree-ext/latest/ostree_ext/container/enum.SignatureSource.html
+type ImageSignature struct {
+	AllowInsecure bool
+	OstreeRemote  string
+}
+
+// bootc.go
+// Synchronously invoke bootc, writing its stdout to our stdout,
+// and gathering stderr into a buffer which will be returned in err
+// in case of error.
+func runBootc(args ...string) error {
+	return runCmdSync("bootc", args...)
+}
+
+func (b *BootcClient) Initialize() error {
+	// Commands like update and rebase need the pull secrets to pull images and manifests,
+	// make sure we get access to them when we Initialize
+	err := useMergedPullSecrets(bootcSystem)
+	if err != nil {
+		return fmt.Errorf("Error while ensuring access to kublet config.json pull secrets: %w", err)
+	}
+
+	return nil
+}
+
+// GetBootedAndStagedImage returns the current booted and staged image found
+func (b *BootcClient) GetBootedAndStagedImage() (booted, staged *BootEntry, err error) {
+	status, err := b.QueryStatus()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	booted = status.GetBootedImage()
+	staged = status.GetStagedImage()
+
+	return
+}
+
+// // GetStatus returns multi-line human-readable text describing system status
+// // Bootc is working on status update: https://github.com/containers/bootc/issues/408
+// func (r *BootcClient) GetStatus() (string, error) {
+// 	output, err := runGetOut("bootc", "status")
+// 	if err != nil {
+// 		return "", err
+// 	}
+
+// 	return string(output), nil
+// }
+
+// GetBootcBootedOSImageURL returns the image URL as well as the image version(for logging) and the ostree commit (for comparisons)
+func (b *BootcClient) GetBootcBootedOSImageURL() (string, string, string, error) {
```

Comment:
Save this suggestion for future discussion: 
Put into a struct - May want to implement this later when we are settled down on what to return here for a useful combo of info. Currently, these three strings are chosen due to similar implementation in rpm-ostree. 
Attach to BootcClient - Doing so because we will eventually separate the go-bindings to a standalone library. When vendoring it back in, BootcClient will be the main and only wrapper for all bootc related methos and status update.

Reply:
I neglected to include my rationale when I wrote the original comment. Whenever one returns multiple values of the same type from a function (not including errors), there's a possibility that one can introduce subtle and difficult-to-track bugs. For example:

```golang
func getUserInfo() (string, string, string) {
    return "Jane Doe", "janedoe@company.com", "password"
}

func storeUserInfo(name, email, password string) {
    fmt.Println("Storing user info for", name)
    // Code that writes info to disk
}

func main() {
    // Let's say that you accidentally transpose the return values like this:
    email, password, name := getThings()

    // But you put them in the correct order for this function.
    storeUserInfo(name, email, password)

    // The password will get printed out; definitely not desirable.
}
```

By comparison, if you have a single composite type, you get this:

```golang
type userInfo struct {
    name string
    email string
    password string
}

func getUserInfo() userInfo {
    return userInfo{
        name: "Jane Doe",
        email: "janedoe@company.com",
        password: "password",
    }
}

func storeUserInfo(user userInfo) {
    fmt.Println("Storing user info for", user.name)
    // Code that writes this to disk
}

func main() {
    // Doing it this way, the order doesn't matter.
    user := getUserInfo()

    storeUserInfo(user)
    // The password will not get printed out in this case because it didn't get transposed.
    // And while there is a way to initialize a struct by implicitly relying
    // on the order of the fields, one can prevent that by adding a lint rule.
}
```

Reply:
Update for rpm-ostree as well :)